Test parser fails on wrong EVMVersion syntax

Input:

```
contract c {
  function f() public pure returns (uint) {
    return 7;
  }
}
// ====
// compileViaYul: also
// EVMVersion: byzantium
// ----
// f(uint256): 7 -> 7
```

It fails with an exception because of the syntax error in EVMVersion:

```
/Users/alex/Projects/solidity/test/boostTest.cpp:125: fatal error: in "semanticTests/wasm": unexpected exception thrown by { try { stringstream errorStream; auto testCase = _testCaseCreator(config); if (testCase->validateSettings(dev::test::Options::get().evmVersion())) switch (testCase->run(errorStream)) { case TestCase::TestResult::Success: break; case TestCase::TestResult::Failure: do { ::boost::unit_test::unit_test_log.set_checkpoint( ::boost::unit_test::const_string( "/Users/alex/Projects/solidity/test/boostTest.cpp", sizeof( "/Users/alex/Projects/solidity/test/boostTest.cpp" ) - 1 ), static_cast<std::size_t>(114) ); ::boost::test_tools::tt_detail:: report_assertion ( (false), (::boost::unit_test::lazy_ostream::instance() << "Test expectation mismatch.\n" + errorStream.str()), ::boost::unit_test::const_string( "/Users/alex/Projects/solidity/test/boostTest.cpp", sizeof( "/Users/alex/Projects/solidity/test/boostTest.cpp" ) - 1 ), static_cast<std::size_t>(114), ::boost::test_tools::tt_detail::CHECK, ::boost::test_tools::tt_detail::CHECK_MSG , 0 ); } while( ::boost::test_tools::tt_detail::dummy_cond() ); break; case TestCase::TestResult::FatalError: do { ::boost::unit_test::unit_test_log.set_checkpoint( ::boost::unit_test::const_string( "/Users/alex/Projects/solidity/test/boostTest.cpp", sizeof( "/Users/alex/Projects/solidity/test/boostTest.cpp" ) - 1 ), static_cast<std::size_t>(117) ); ::boost::test_tools::tt_detail:: report_assertion ( (false), (::boost::unit_test::lazy_ostream::instance() << "Fatal error during test.\n" + errorStream.str()), ::boost::unit_test::const_string( "/Users/alex/Projects/solidity/test/boostTest.cpp", sizeof( "/Users/alex/Projects/solidity/test/boostTest.cpp" ) - 1 ), static_cast<std::size_t>(117), ::boost::test_tools::tt_detail::CHECK, ::boost::test_tools::tt_detail::CHECK_MSG , 0 ); } while( ::boost::test_tools::tt_detail::dummy_cond() ); break; } } catch (boost::exception const& _e) { do { ::boost::unit_test::unit_test_log.set_checkpoint( ::boost::unit_test::const_string( "/Users/alex/Projects/solidity/test/boostTest.cpp", sizeof( "/Users/alex/Projects/solidity/test/boostTest.cpp" ) - 1 ), static_cast<std::size_t>(123) ); ::boost::test_tools::tt_detail:: report_assertion ( (false), (::boost::unit_test::lazy_ostream::instance() << "Exception during extracted test: " << boost::diagnostic_information(_e)), ::boost::unit_test::const_string( "/Users/alex/Projects/solidity/test/boostTest.cpp", sizeof( "/Users/alex/Projects/solidity/test/boostTest.cpp" ) - 1 ), static_cast<std::size_t>(123), ::boost::test_tools::tt_detail::CHECK, ::boost::test_tools::tt_detail::CHECK_MSG , 0 ); } while( ::boost::test_tools::tt_detail::dummy_cond() ); } }
```

--- a/test/TestCase.cpp
+++ b/test/TestCase.cpp
@@ -186,7 +186,7 @@ bool EVMVersionRestrictedTestCase::validateSettings(langutil::EVMVersion _evmVer
 	versionString = versionString.substr(versionBegin);
 	std::optional<langutil::EVMVersion> version = langutil::EVMVersion::fromString(versionString);
 	if (!version)
-		throw runtime_error("Invalid EVM version: \"" + versionString + "\"");
+		BOOST_THROW_EXCEPTION(runtime_error{"Invalid EVM version: \"" + versionString + "\""});
 
 	if (comparator == ">")
 		return _evmVersion > version;
@@ -201,6 +201,5 @@ bool EVMVersionRestrictedTestCase::validateSettings(langutil::EVMVersion _evmVer
 	else if (comparator == "!")
 		return !(_evmVersion == version);
 	else
-		throw runtime_error("Invalid EVM comparator: \"" + comparator + "\"");
-	return false; // not reached
+		BOOST_THROW_EXCEPTION(runtime_error{"Invalid EVM comparator: \"" + comparator + "\""});
 }

